backend/git_real/models.py

```diff
@@ -179,25 +178,28 @@ class Meta:
     @classmethod
     def create_or_update_from_github_api_data(cls, repo, request_body):
 
-        head_commit = request_body["head_commit"]
-        head_commit_url = head_commit["url"]
-        ref = request_body["ref"]
-        defaults = {
-            "commit_timestamp": head_commit["timestamp"],
-            "author_github_name": head_commit["author"]["username"],
-            "committer_github_name": head_commit["committer"]["username"],
-            "message": head_commit["message"],
-            "pusher_username": request_body["pusher"]["name"],
-            "pushed_at_time": github_timestamp_int_to_tz_aware_datetime(
-                int(request_body["repository"]["pushed_at"])
-            ),
-        }
-
-        instance, _ = cls.get_or_create_or_update(
-            repository=repo,
-            head_commit_url=head_commit_url,
-            ref=ref,
-            defaults=defaults,
-            overrides=defaults,
-        )
-        return instance
+        if not request_body["head_commit"]:
```

Comment:
if the event is a Push and the head_commit is empty then return none.

Explicit is better than implicit.
Keep your intention clear. This if statement only matters with push events. 

Why does this matter? Imagine we add a new event type, eg "Repo created". When a repo gets created then there is no head commit (since nothing is committed yet). Now the webhook will hneed to be altered.

If one of our other events has an error because we expected a commit id and didn't get one, then the code should not fail silently. It should fail loudly so that we know there is a problem.